backends/hospital-service/main.bal

```diff
@@ -0,0 +1,286 @@
+        var {SSN: ssn, symptoms, treatments} = patientDetails;
+        lock {
+            if !patients.hasKey(ssn) {
+                return {body: "unknown patient"};
+            }
+        }
+
+        lock {
+            PatientRecord {symptoms: symptomsRecord, treatments: treatmentsRecord} = patientRecords.get(ssn);
+            string time = regexp:split(re `T`, time:utcToString(time:utcNow()))[0];
+            symptomsRecord[time] = symptoms;
+            treatmentsRecord[time] = treatments;
+        }
+        return {status: "update successful"};
+    }
+
+    resource function get [string hospital]/categories/patient/[string SSN]/getrecord()
+            returns PatientRecord|http:NotFound {
+        lock {
+            if patientRecords.hasKey(SSN) {
+                return patientRecords.get(SSN).clone();
+            }
+        }
+        return {body: "unknown patient entry"};
+    }
+
+    resource function get [string hospital]/categories/patient/appointment/[int appointment_id]/discount()
+            returns Status|http:NotFound|error {
+        lock {
+            if appointments.hasKey(appointment_id) {
+                string dob = appointments.get(appointment_id).patient.dob;
+                string yob = regexp:split(re `-`, dob)[0];
+                string currYear = regexp:split(re `-`, time:utcToString(time:utcNow()))[0];
+                int age = check int:fromString(currYear) - check int:fromString(yob);
+                return {status: (age < 12 || age > 55).toString()};
+            }
+        }
+        return {body: "unknown appointment ID"};
+    }
+
+    resource function post [string hospital]/categories/admin/doctor/newdoctor(Doctor & readonly doctor)
+            returns Status|http:NotFound => addDoctor(doctor);
+}
+
+isolated function getDoctor(string name) returns (Doctor & readonly)? {
+    lock {
+        return doctors.hasKey(name) ? doctors.get(name) : ();
+    }
+}
+
+isolated function getDoctorsByCategory(string category) returns Doctor[] & readonly {
+    lock {
+        return from Doctor & readonly doctor in doctors 
+                where doctor.category == category
+                select doctor;
+    }
+}
+
+isolated function getAppointment(int appointmentId) returns (Appointment & readonly)|http:NotFound {
+    lock {
+        if appointments.hasKey(appointmentId) {
+            return appointments.get(appointmentId).cloneReadOnly();
+        }
+    }
+    return {body: "unknown appointment ID"};
+}
+
+isolated function addDoctor(Doctor & readonly doctor) returns Status|http:NotFound {
+    final var {name, category} = doctor;
+    lock {
+        // if !categories.some(existingCategory => existingCategory == category) { // crashes
+        if categories.indexOf(category) is () {
+            categories.push(category);
+        }
+    }
+
+
+    lock {
+        if getDoctor(name) !is () {
+            return {status: "doctor already exists in the system"};
```

Comment:
Here we are sending an error. Shouldn't this be a HTTP error instead of a Status type?

Reply:
https://github.com/ballerina-guides/integration-tutorials/pull/2#discussion_r1302774640